lib/internal/modules/esm/loader.js

```diff
@@ -225,8 +240,50 @@ class DefaultModuleLoader {
    * @returns {Promise<ModuleExports>}
    */
   async import(specifier, parentURL, importAssertions) {
+    const { internalCache, serializedModuleRequest } =
+      this.#resolveCache.getSerialized(specifier, parentURL, importAssertions);
+    const removeCache = () => {
+      // Remove the cache entry if the import fails.
+      delete internalCache[serializedModuleRequest];
```

Comment:
This only removes the entry _relative to its parent,_ right? Since `internalCache` is defined for each `parentURL`? Shouldn’t failures remove the cache entry everywhere? Is there a way to define the resolve cache such that it’s not a map of maps, but closer to a straightforward key-value cache?

Reply:
> Shouldn’t failures remove the cache entry everywhere?

No, if `import("./file.js")` fails in `a/entry.js`, that shouldn't invalidate the cache entry for `import("./file.js")` in `b/other_file.js`.

Reply:
Okay this makes sense. Perhaps it should be named something like `parentURLImports` to be clearer what it is? And like all the caches, a comment explaining the keys and values would help.